File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.go
func groupAnagrams(strs []string) [][]string {
    //empty map that will store final groups
    //loop each word
    //each word - sorted key
    //use key to append original word to list

	freqMap := make(map[string][]string)
	for _, word := range strs {
		runes := []rune(word)
		sort.Slice(runes, func(i, j int) bool {
			return runes[i] < runes[j]
		})
        key := string(runes)
        freqMap[key] = append(freqMap[key], word)
	}
    finalList := make([][]string, 0, len(freqMap))
    for _, anagrams := range freqMap {
    finalList = append(finalList, anagrams)
    }  
    return finalList
}

File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> ump;
        for (string s : strs) {
            string key = s;
            sort(key.begin(), key.end());
            ump[key].push_back(s);
        }
        for (auto& iter : ump) {
            ans.push_back(iter.second);
        }
        return ans;
    }
};